pages/search/search.jsx

```diff
@@ -1,48 +1,47 @@
 import React from 'react';
 import { browserHistory } from 'react-router';
 import classNames from 'classnames';
-import ProjectList from '../../components/project-list/project-list.jsx';
+import DebounceInput from 'react-debounce-input';
+import ProjectCard from '../../components/project-card/project-card.jsx';
+import Service from '../../js/service.js';
+import Utility from '../../js/utility.js';
 
 export default React.createClass({
+  numProjectsInBatch: 24, // make sure this number is divisible by 2 AND 3 so rows display evenly for different screen sizes
   getInitialState() {
     return {
-      searchQuery: ``
+      displayBatchIndex: 1,
+      keywordBeingSearched: ``,
+      keywordSearched: ``,
+      entriesMatched: []
     };
   },
   componentWillReceiveProps(nextProps) {
-    // this makes sure searchQuery is updated accordingly when
-    // browser's back / forward button is triggered
-    this.setSearchQuery(nextProps.location.query.keyword);
+    // when window.history.back() or windows.history.forward() is triggered
+    // (e.g., clicking on browser's back / forward button)
+    // we want to make sure search result gets updated accordingly
+    let searchKeyword = nextProps.location.query.keyword;
+    this.searchByQueryInUrl(searchKeyword);
   },
   componentDidMount() {
-    this.setSearchQuery(this.props.location.query.keyword);
+    let queryKeyword = this.props.location.query.keyword;
+    this.searchByQueryInUrl(queryKeyword);
   },
-  setSearchQuery(searchQuery) {
-    if (searchQuery !== this.searchInput.value) {
-      this.searchInput.focus();
-      this.searchInput.value = searchQuery ? decodeURIComponent(searchQuery) : ``;
-      this.setState({searchQuery: searchQuery});
-    }
-  },
-  clearSearch() {
-    this.searchInput.value = ``;
-    this.searchInput.blur();
-  },
-  handleInputBlur() {
-    this.updateSearchQuery();
-  },
-  handleInputKeyUp(event) {
-    if (event.keyCode === 27) { // escape key
-      this.clearSearch();
+  searchByQueryInUrl(searchQueryInUrl) {
+    let inputBoxValue = this.refs.searchInput.state.value;
+
+    if (searchQueryInUrl !== inputBoxValue) {
+      // make sure input box value is updated with searchQueryInUrl
+      // then use searchQueryInUrl as the search param to fetch data from Pulse API
+      this.refs.searchInput.setState({value: searchQueryInUrl ? decodeURIComponent(searchQueryInUrl) : ``});
+      this.setState({
+        keywordBeingSearched: searchQueryInUrl
```

Comment:
What's the purpose of this state variable? It seems like it's only checked on line 85, which could just read the value of the input box directly, no?